src/Api/State/PersonalDataStateProcessor.php

```diff
@@ -54,6 +55,14 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             return $this->persistProcessor->process($data, $operation, $uriVariables, $context);
         }
 
+        if ($data instanceof DonneePersonnelle && str_starts_with((string) $context['operation']->getName(), (string) ApiRoutes::TOGGLE_VISIBILITY)) {
+            if (!$user instanceof User) {
```

Comment:
`str_starts_with((string) $context['operation']->getName(), (string) ApiRoutes::TOGGLE_VISIBILITY)` Maybe we could extract this.
`ApiRoutes.php` could be an helper with operations and methods in order to achieve this